lib/new_relic/agent/threading/internal_thread.rb

```diff
@@ -0,0 +1,14 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+module NewRelic
+  module Agent
+    module Threading
+      # This class is used internally by the agent to prevent agent threads from sharing a state with application threads
+      class InternalThread < ::Thread
```

Comment:
I love this solution. Do you think this new class could be used in the future to help accommodate a blocklist of app threads that should not be monitored? I'm thinking the agent could use this new class for those types of threads. If so, I propose that this class be renamed and the comment updated so that we go with something more generic than "internal". Perhaps something like "NonMonitored", "Unmonitored", "Ignored", "Unreported", etc.

Reply:
Are you thinking that this should be a public api class like we did for TracedThread?

Reply:
Yep, exactly. But after some quick sketching out of how it would work, I've changed my mind.

It's a lot easier to rename and expose a private API later than it is to make something public now prematurely before it's wired up to anything.

The more I think about it, the more I'd prefer to work out a long term solution that allows an app to simply use `::Thread` but tell us whether or not it should be traced. Perhaps by monkeypatching `::Thread.initialize`. That would allow the app's own source code to continue to use `::Thread` instead of a NR proprietary class name.

I withdraw my suggestion for a rename. I think we should leave things as is for now before we work out the idea more.